### PR TITLE
Lesson1

### PR DESCRIPTION
**Pull request by KarachunVladislav01**
First task for check

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+      type="text/javascript"
+    ></script>
+    <link rel="stylesheet" href="./styles/style.css" />
+    <script type="module" src="./script.js"></script>
+    <title>WebixLayout</title>
+  </head>
+  <body>
+    <span
+      >The software is provided by<a href="https://docs.webix.com">webix.com</a>
+      https://docs.webix.com. Allrights reseved (c)</span
```
Comment (notheheanymore) [inline index.html]: You have two labels like this on the page.
image

File: index.html [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link
+      rel="stylesheet"
+      href="https://cdn.webix.com/edge/webix.css"
+      type="text/css"
+    />
+    <link
+      rel="stylesheet"
+      href="https://cdn.materialdesignicons.com/2.7.94/css/materialdesignicons.css"
```
Comment (Zwillinge) [inline index.html]: no need to load extra icon pack if there're built-in icons in the lib.

File: script.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      minWidth: 200
+    },
+    {},
+    {
+      view: "button",
+      type: "icon",
+      icon: "mdi mdi-account",
+      label: "Profile",
+      gravity: 0.2,
+      minWidth: 200
```
Comment (notheheanymore) [inline script.js]: Not consistent width of button. You can check it on hover
image

File: index.html [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+    <script
+      src="https://cdn.webix.com/edge/webix.js"
+      type="text/javascript"
+    ></script>
+    <link rel="stylesheet" href="./styles/style.css" />
+    <script type="module" src="./script.js"></script>
+    <title>WebixLayout</title>
+  </head>
+  <body>
+    <span
```
Comment (Zwillinge) [inline index.html]: do you really need this span in html?
its duplicate is already included into webix layout

File: script.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+const form = {
+  view: "form",
+  minWidth: 150,
+  maxWidth: 250,
+  elements: [
+    { template: "EDIT FILMS", type: "section" },
+    {
+      margin: 20,
+      cols: [{ view: "label", label: "Title", width: 70 }, { view: "text" }]
```
Comment (Zwillinge) [inline script.js]: bad practice. you need to use a Text component and set the label property for it

File: script.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      cols: [{ view: "label", label: "Rating", width: 70 }, { view: "text" }]
+    },
+    {
+      margin: 20,
+      cols: [{ view: "label", label: "Votes", width: 70 }, { view: "text" }]
+    },
+    {
+      margin: 20,
+      cols: [
+        { view: "button", label: "Add new", css: "form-button--add" },
```
Comment (Zwillinge) [inline script.js]: use preset CSS class, to make the button primary

File: script.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+};
+
+const dataTable = {
+  view: "datatable",
+  minWidth: 540,
+  data: filmsSet,
+  autoConfig: true
+};
+
+const sideMenu = {
```
Comment (notheheanymore) [inline script.js:86]: Menu should not fill half page. It should be smaller
image

File: script.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  maxHeight: 40
+};
+
+webix.ready(function() {
+  webix
+    .ui({
+      id: "layout",
+      rows: [header, content, footer]
+    })
+    .show();
```
Comment (Zwillinge) [inline script.js]: for layouts there's no need to call .show explicitely.

File: script.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      cols: [
+        { view: "button", label: "Add new", css: "form-button--add" },
+        { view: "button", label: "Clear" }
+      ]
+    },
+    {}
+  ]
+};
+
+const dataTable = {
```
Comment (notheheanymore) [inline script.js:78]: Datatable should not have horizontal scroll

**Review by notheheanymore — CHANGES_REQUESTED** [review]

File: script.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
+import { filmsSet } from "./data/test-data.js";
+
+const header = {
+  css: "header",
```
Comment (Zwillinge) [inline script.js]: use 'native' webix styling for toolbar instead of custom css
like this: https://snippet.webix.com/stuf64ow

**Review by Zwillinge — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (KarachunVladislav01) [reply index.html]: forgot to clean,
fixed it.

File: script.js (hunk 3 above)
Comment (KarachunVladislav01) [reply script.js]: consistent with what?
made less.

File: script.js (hunk 7 above)
Comment (KarachunVladislav01) [reply script.js:86]: fixed it.

File: script.js (hunk 9 above)
Comment (KarachunVladislav01) [reply script.js:78]: fixed it.

File: index.html (hunk 2 above)
Comment (KarachunVladislav01) [reply index.html]: No i start with mat, but after i found webix icons and forgot change that.
fixed it.

File: index.html (hunk 4 above)
Comment (KarachunVladislav01) [reply index.html]: No, forgot clean it,
fixed it.

File: script.js (hunk 5 above)
Comment (KarachunVladislav01) [reply script.js]: understood,
fixed it.

File: script.js (hunk 6 above)
Comment (KarachunVladislav01) [reply script.js]: fixed it.

File: script.js (hunk 8 above)
Comment (KarachunVladislav01) [reply script.js]: Ok
fixed it.

File: script.js (hunk 10 above)
Comment (KarachunVladislav01) [reply script.js]: oh, thanks,
fixed it.
Made all the changes, you can check again

**Review by notheheanymore — APPROVED** [review]

File: script.js (hunk 5 above)
Comment (Zwillinge) [reply script.js]: not really - you don't need to wrap the text field into the `cols` layout as it is the only element in it

Comment (KarachunVladislav01) [reply script.js]: Did it

**Review by Zwillinge — APPROVED** [review]

**Pull request merged**